### PR TITLE
Feature/homework 7

### PR DESCRIPTION
**Pull request by ikushlianski**
# UNIT TESTS AND CONFIG

### TASK 7.1

- Add unit tests for User entity controller methods using Jest library
- Add unit tests for Group entity controller methods using Jest

Commands:
- `npm run test` to run tests
- `npm run coverage` to see coverage % in console and also in `/coverage` directory in project root (`index.html` file)

### TASK 7.2

Pass configs to the application using environment variables with `dotenv` package

> Note: I started using `dotenv` from module 2, so this task had already been done before

### CODE REVIEW

File: src/entities/group/group.controller.spec.ts [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+  describe('getGroups', () => {
+    it('should call group service with custom limit if it is in query params', async () => {
+      req.url = 'http://localhost:3000/api/users?limit=2';
+
+      jest
+        .spyOn(groupService, 'getAll')
+        .mockImplementation(() => Promise.resolve([]));
+      await groupController.getGroups(req, res);
+
+      expect(groupService.getAll).toHaveBeenCalledWith(2);
```
Comment (RockMinsk) [inline src/entities/group/group.controller.spec.ts:37]: It that correct that groupsService.getAll called twice?

File: src/entities/group/group.controller.spec.ts [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+
+      const group = new GroupModel();
+
+      jest
+        .spyOn(groupService, 'getById')
+        .mockImplementation(() => Promise.resolve(group));
+
+      await groupController.getGroupById(req, res);
+
+      expect(res.send).toHaveBeenCalledWith(group);
```
Comment (RockMinsk) [inline src/entities/group/group.controller.spec.ts:112]: Also it's good to check here and in other places that dependent service is called and called only once, something like this:
`expect(groupService.getById).toBeCalledTimes(1);`

File: src/entities/user/user.controller.spec.ts [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+        Promise.resolve({
+          ...req.body,
+          id: '1',
+          isDeleted: true,
+        }),
+      );
+
+      await softDeleteUser(req, res);
+
+      expect(userService.softDelete).toHaveBeenCalledWith('1');
```
Comment (RockMinsk) [inline src/entities/user/user.controller.spec.ts:254]: Expect for message `"Deleted user ${userId}"` is missing.

**Review by RockMinsk — APPROVED** [review]

File: src/entities/group/group.controller.spec.ts (hunk 1 above)
Comment (ikushlianski) [reply src/entities/group/group.controller.spec.ts:37]: it's `toHaveBeenCalledWith`, not `toHaveBeenCalledTimes` :)

File: src/entities/group/group.controller.spec.ts (hunk 2 above)
Comment (ikushlianski) [reply src/entities/group/group.controller.spec.ts:112]: agree

File: src/entities/user/user.controller.spec.ts (hunk 3 above)
Comment (ikushlianski) [reply src/entities/user/user.controller.spec.ts:254]: added

**Pull request merged**